### ISSUE TITLE
`transition:name` can't use Chinese 

### ISSUE THREAD

**Issue by liruifengv** [withastro/astro#9803]
### Astro Info

```block
Astro                    v4.1.0
Node                     v20.9.0
System                   Windows (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/tailwind
                         @astrojs/react
                         @astrojs/sitemap
                         astro-expressive-code
                         @astrojs/mdx
```

### If this issue only occurs in one browser, which browser is a problem?

Chrome

### Describe the Bug

Set a element `transition:name` to `transition:name={name}`
The name is Chinese text. 

Read Astro code:

```ts
// Ensure animationName is a valid CSS identifier
function toValidIdent(name: string): string {
	return name.replace(/[^a-zA-Z0-9\-\_]/g, '_').replace(/^\_+|\_+$/g, '');
}
```

This will cause the Chinese name to be replaced with empty, resulting in multiple duplicates of the name.

![image](https://github.com/withastro/astro/assets/25167721/153fa143-59de-40ab-b438-8ead66dc40da)

## My website as example:
https://liruifengv.com/tags/

![image](https://github.com/withastro/astro/assets/25167721/489c5b47-c929-4d37-95c9-eab0f9d87f9b)

I think the same will be true for Japanese and Korean, etc.

### What's the expected result?

Expect Chinese to be used correctly as transition:name.

CSS identifiers can be Chinese、Japanese and Korean, etc.

### Link to Minimal Reproducible Example

https://liruifengv.com/tags/

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by liruifengv** [comment]
This is causing me a lot of problems and causing my site animations to fail.

I checked the commit history and found that this PR did a replacement process for `transition:name`.

Why do this, I think it's a break change.

https://github.com/withastro/astro/pull/8207

**Comment by martrapp** [comment]
As you can see, I'm having trouble choosing the right label here. This is definitely not an edge case, but it also does not contradict the documented behavior. 

The browser only accepts a limited [range of characters](https://developer.mozilla.org/en-US/docs/Web/CSS/custom-ident) for view-transition-names. 
The only valid unescaped characters are A-Z, a-z, 0-9, and \-_.

**Comment by liruifengv** [comment]
Maybe we can encode it using `encodeURIComponent` before.

**Comment by liruifengv** [comment]
Yeah. We need give a warning when change the user passed value.

**Comment by martrapp** [comment]
> Maybe we can encode it using encodeURIComponent before.

This would yield %E5%BC%80%E6%BA%90 which is also illegal, but I know what you mean and would agree if we find a robust way to do so.

**Comment by delucis** [comment]
I took a quick look and didn’t immediately spot a definitive CSS ident escaping tool, but yes I’d be +1 on auto-escaping if possible and definitely warning in case someone is using the name elsewhere in their CSS.

**Comment by martrapp** [comment]
This might be a bit bumpy, but I'll take a look.
Or will someone else volunteer?

**Comment by liruifengv** [comment]
I'll also take a look.

**Comment by ccbikai** [comment]
Is it possible to use `encodeURIComponent('中文').replaceAll('%', '_')`?

Although longer, they are all legal characters

**Comment by martrapp** [comment]
> Is it possible to use encodeURIComponent('中文').replaceAll('%', '_')?

true, but it would be rather inconvenient to decode. When using `\5F00\6E90`, you get the original text in the browsers development tools:
![image](https://github.com/withastro/astro/assets/94928215/6edc9389-df5e-429f-8c2b-afc5c442c0ff)

**Comment by liruifengv** [comment]
I just made a PR. Can you help me with it? @delucis @martrapp 

https://github.com/withastro/astro/pull/9822

**Comment by martrapp** [comment]
Great! Thanks a lot, happy to do so!
